src/Plugins/RunawayProcessKiller/RunawayProcessKillerExtension.cs

```diff
@@ -193,6 +194,68 @@ public override void Configure(IWinSWConfiguration descriptor, XmlNode node)
             this.CheckWinSWEnvironmentVariable = checkWinSWEnvironmentVariable is null ? true : bool.Parse(checkWinSWEnvironmentVariable);
         }
 
+        public override void Configure(IWinSWConfiguration descriptor, YamlExtensionConfiguration config)
+        {
+            var dict = config.GetSettings();
+
+            this.Pidfile = (string)dict["pidfile"];
+            this.StopTimeout = TimeSpan.FromMilliseconds(int.Parse((string)dict["stopTimeOut"]));
+            this.StopParentProcessFirst = bool.Parse((string)dict["StopParentFirst"]);
+
+            try
+            {
+                this.CheckWinSWEnvironmentVariable = bool.Parse((string)dict["checkWinSWEnvironmentVariable"]);
+            }
+            catch
+            {
+                this.CheckWinSWEnvironmentVariable = true;
+            }
+        }
+
+        public override void Configure(IWinSWConfiguration descriptor, object yamlObject)
+        {
+            if (!(yamlObject is Dictionary<object, object> dict))
+            {
+                // TODO : throw ExtensionException
+                throw new InvalidDataException("Cann't configure");
```

Comment:
```suggestion
                throw new InvalidDataException("Cannot configure the extension: Input YAML object is not a dictionary");
```

Reply:
The code is not throwing this anymore. I wrap the YAML object with a class?